### DISCUSSION TITLE
Any thoughts on moving from npm in favor of something like pnpm, yarn, etc.

### DISCUSSION THREAD

**Discussion by wreiske** [meteor/meteor#13316] [Feature Requests]
I've been experimenting with pnpm lately and thought it would be good to start a discussion here about it.

https://pnpm.io/pnpm-vs-npm
https://pnpm.io/feature-comparison

- Fast. Up to 2x faster than the alternatives (see [benchmark](https://github.com/pnpm/pnpm?tab=readme-ov-file#benchmark)).
- Efficient. Files inside node_modules are linked from a single content-addressable storage.
- [Great for monorepos](https://pnpm.io/workspaces).
- Strict. A package can access only dependencies that are specified in its package.json.
- Deterministic. Has a lockfile called pnpm-lock.yaml.
- Works as a Node.js version manager. See [pnpm env use](https://pnpm.io/cli/env).
- Works everywhere. Supports Windows, Linux, and macOS.
- Battle-tested. Used in production by teams of [all sizes](https://pnpm.io/users) since 2016.

https://www.reddit.com/r/node/comments/144xqd8/is_pnpm_really_leaves_up_to_its_hype_are_yarn_npm/
https://refine.dev/blog/pnpm-vs-npm-and-yarn/
https://www.syncfusion.com/blogs/post/pnpm-vs-npm-vs-yarn

1. What are your thoughts on supporting pnpm out of the box for future versions of meteor?
2. Should meteor move completely to pnpm as it's built in node module package manager? e.g., `meteor npm` would become an alias of `meteor pnpm` and use pnpm.

Might be cool to get included on pages like this: https://pnpm.io/workspaces#usage-examples to get better visibility of meteor.

**Comment by StorytellerCZ** [comment]
NPM needs to remain supported as it is the default Node package manager. Yarn is supported as well. Adding pnpm support is fine with me.
